operations/agent-flow-mixin/dashboards/opentelemetry.libsonnet

```diff
@@ -0,0 +1,174 @@
+        label_values(agent_component_controller_running_components{cluster="$cluster", namespace="$namespace"}, instance)
+      |||),
+    ]) +
+    dashboard.withPanelsMixin([
+      // "Receivers for traces" row
+      (
+        panel.new('Receivers for traces [otelcol.receiver]', 'row') +
+        panel.withPosition({ h: 1, w: 24, x: 0, y: 0 })
+      ),
+      (
+        panel.new(title='Accepted spans', type='timeseries') +
+        panel.withDescription(|||
+         Number of spans successfully pushed into the pipeline.
+       |||) +
+        panel.withPosition({ x: 0, y: 0, w: 8, h: 10 }) +
+        panel.withQueries([
+          panel.newQuery(
+            expr=|||
+              rate(receiver_accepted_spans_ratio_total{cluster="$cluster", namespace="$namespace", instance=~"$instance"}[$__rate_interval])
+            |||,
+            //TODO: How will the dashboard look if there is more than one receiver component? The legend is not unique enough?
+            legendFormat='{{ pod }} / {{ transport }}',
+          ),
+        ])
+      ),
+      (
+        panel.new(title='Refused spans', type='timeseries') +
+        stackedPanelMixin +
+        panel.withDescription(|||
+          Number of spans that could not be pushed into the pipeline.
+        |||) +
+        panel.withPosition({ x: 8, y: 0, w: 8, h: 10 }) +
+        panel.withQueries([
+          panel.newQuery(
+            expr=|||
+             rate(receiver_refused_spans_ratio_total{cluster="$cluster", namespace="$namespace", instance=~"$instance"}[$__rate_interval])
+            |||,
+            legendFormat='{{ pod }} / {{ transport }}',
+          ),
+        ])
+      ),
+      (
+        panel.newHeatmap('RPC server duration (traces)') +
+        panel.withUnit('milliseconds') +
+        panel.withDescription(|||
+          The duration of inbound RPCs.
+        |||) +
+        panel.withPosition({ x: 16, y: 0, w: 8, h: 10 }) +
+        panel.withQueries([
+          panel.newQuery(
+            expr='sum by (le) (increase(rpc_server_duration_milliseconds_bucket{cluster="$cluster", namespace="$namespace", instance=~"$instance", rpc_service="opentelemetry.proto.collector.trace.v1.TraceService"}[$__rate_interval]))',
+            format='heatmap',
+            legendFormat='{{le}}',
+          ),
+        ])
+      ),
+
+      // "Batching" row
+      (
+        panel.new('Batching [otelcol.processor.batch]', 'row') +
+        panel.withPosition({ h: 1, w: 24, x: 0, y: 10 })
+      ),
+      (
+        panel.newHeatmap('Number of units in the batch') +
+        panel.withDescription(|||
+          Number of units in the batch
+        |||) +
+        panel.withPosition({ x: 0, y: 10, w: 8, h: 10 }) +
+        panel.withQueries([
+          panel.newQuery(
+            expr='sum by (le) (increase(processor_batch_batch_send_size_ratio_bucket{cluster="$cluster", namespace="$namespace", instance=~"$instance"}[$__rate_interval]))',
```

Comment:
Here we also get negative values:
![Screenshot 2024-01-17 at 16 29 09](https://github.com/grafana/agent/assets/5834788/fec874ce-eb99-4abb-b1fc-dedfb19bfc81)

Reply:
As discussed offline: we're still a little unclear on what the actual underlying metric is about (`ratio`? of what?). Does the 220 upper-limit value here make sense compared to the rate of data we're seeing in the actual environment>

We can take a look, but maybe we could remove this panel for now and put it back on after we've got a better understanding of what we're looking at here.

Reply:
I think those weird buckets are only visible in my locally running Grafana and not in the actual environment (even for the same data in Mimir), so I'll assume that the dashboard is ok for now and will merge it as is. We can improve it over time.